src/main/resources/templates/cart.html

```diff
@@ -51,31 +51,29 @@ <h1>我的购物车</h1>
                 <th>商品图片</th>
                 <th>商品描述</th>
                 <th>商品单价</th>
-                <th>用户操作</th>
+                <th>商品数量</th>
+                <th>选择</th>
             </tr>
-            <tr th:each="sItem:${cartItems}">
-                <td th:text="${sItem.getCurrentItem().getName()}"></td>
+            <tr th:each="sItem:${cartItems}" id="product">
+                <td id="ItemName" th:text="${sItem.getCurrentItem().getName()}"></td>
                 <td><img th:src="${sItem.getCurrentItem().getImage()}"></td>
                 <td th:text="${sItem.getCurrentItem().getDescr()}"></td>
                 <td class="sPrice" th:text="${sItem.getCurrentItem().getPrice()}"></td>
-                <td>
-                    <form class="sItemCartUpdateForm">
-                        <input type="hidden" name="sItemID" id="sItemID" th:value="${sItem.getCurrentItem().getId()}">
-                        <label for="sItemNo">当前数量</label>
-                        <input type="number" name="sItemNo" id="sItemNo" th:value="${sItem.getItemNum()}">
-                        <button onclick="deleteItemFromCart(this)">删除当前商品</button>
-                        <label for="sItemChecked">勾选以结算</label>
-                        <input type="checkbox" id="sItemChecked" name="sItemChecked">
-                    </form>
+                <td >
+                    <input type="hidden" name="sItemID" id="sItemID" th:value="${sItem.getCurrentItem().getId()}">
+                    <label for="sItemNo">当前数量</label>
+                    <input class="ItemNo" type="number" name="sItemNo" id="sItemNo" th:value="${sItem.getItemNum()}">
                 </td>
+                <td><input type="checkbox" id="sItemChecked" name="sItemChecked"><label for="sItemChecked">选中商品</label></td>
             </tr>
         </table>
-        <button onclick="uploadCart()">更新购物车内的物品和数量到服务器</button>
-        <button onclick="updateFinalPrice()">点击更新勾选商品总价</button>
-        <button onclick="submitOrder()">将勾选商品提交订单</button>
+        <button onclick="updateCart()">更新商品数量</button>
+        <button onclick="updateFinalPrice()">获取商品总价</button>
+        <button onclick="submitOrder()">提交商品订单</button>
+        <button onclick="deleteItemFromCart(this)">删除当前商品</button>
```

Comment:
Can you make sure delete the correct item when there's multiple items in the cart?

That's why I put this delete button inside each line instead of the bottom of final form.

Reply:
before delete item checkbox must be selected,then will submit request to api.after my test,it will correctlly submit a request